client/orchestrator-chain-rpc-interface/src/lib.rs

```diff
@@ -318,4 +319,17 @@ impl OrchestratorChainInterface for OrchestratorChainRpcClient {
     async fn finalized_block_hash(&self) -> OrchestratorChainResult<PHash> {
         self.request("chain_getFinalizedHead", rpc_params![]).await
     }
+
+    async fn data_preserver_active_assignment(
```

Comment:
@tmpolaczyk how well will this plug with your changes?

Reply:
Note that now "data preserver mode" is no longer implemented using a subcommand but by a normal CLI argument. You'll thus probably have to add this argument to your subcommand.

Reply:
Yeah, basically we need a `rpc_provider_mode_solochain` function